app/src/androidTest/java/com/github/campus_capture/bootcamp/MainActivityRuleFragmentTest.java

```diff
@@ -21,20 +36,63 @@
 @RunWith(AndroidJUnit4.class)
 public class MainActivityRuleFragmentTest {
 
+    static FirebaseDatabase database = null;
+
     @Rule
     public ActivityScenarioRule<MainActivity> testRule = new ActivityScenarioRule<>(MainActivity.class);
 
-    @Ignore("We can only test it once we have a proper mock DB we can inject")
+    @BeforeClass
+    public static void init_firebase_emulator() {
+
+        try{
+            AppContext context = (AppContext)ApplicationProvider.getApplicationContext();
+            database = context.getFirebaseDB();
+            database.useEmulator("127.0.0.1", 9000);
+
+            //check for connection
+            DatabaseReference connectedRef = FirebaseDatabase.getInstance().getReference(".info/connected");
+            connectedRef.addValueEventListener(new ValueEventListener() {
+                @Override
+                public void onDataChange(@NonNull DataSnapshot snapshot) {
+                    boolean connected = snapshot.getValue(Boolean.class);
+                    if (connected) {
+                        Log.d(TAG, "connected");
+                    } else {
+                        Log.d(TAG, "not connected");
+                    }
+                }
+
+                @Override
+                public void onCancelled(@NonNull DatabaseError error) {
+                    Log.w(TAG, "Listener was cancelled");
+                }
+            });
+
+        } catch (Exception e) {
+            Log.e("Error init DB", e.toString());
+        }
+    }
+
+    @Before
+    public void clear_firebase_database(){
+        database.getReference().setValue(null);
+    }
+
+    @Ignore //TODO comment out once firebase emulator is setup on CI
     @Test
     public void RulesFragmentShowTheRightContent() {
```

Comment:
This test can be removed later since we've decided that the rules fragment should not fetch the rules from the DB and instead read them locally (like from some text file)

Reply:
Yes, this is a thing we can keep in mind.